### PR TITLE
test aio : Add support for for Quark_se family processors.

### PR DESCRIPTION
**Pull request by youvedeep**
tests: aio: port AIO cases to run on more platforms   : This patch is ported from gerrit to github Quipeiyang is Author of this patch.
And reset 2 patches address the bugs as described by Jira https://jira.zephyrproject.org/browse/ZEP-1927 and ZEP-1438.

### CODE REVIEW

**Comment by nashif** [comment]
Please add more details to the PR title and description.

**Review by nashif — DISMISSED** [review]
please add more details to the PR description and title

File: tests/drivers/aio/aio_basic_api/src/test_callback.c [hunk 1]
```diff
@@ -1,54 +1,149 @@
...
 
-	/* update the AIN input to trigger state */
-	gpio_pin_write(gpio_dev, PIN_OUT,
-		       (polarity == AIO_CMP_POL_RISE) ? 0 : 1);
 	k_sleep(100);
+	TC_PRINT("[Debug]: before clear cb_cnt\n");
 	cb_cnt = 0;
-	gpio_pin_write(gpio_dev, PIN_OUT,
-		       (polarity == AIO_CMP_POL_RISE) ? 1 : 0);
+	TC_PRINT("[Debug]: after clear cb_cnt\n");
```
Comment (andrewboie) [inline tests/drivers/aio/aio_basic_api/src/test_callback.c]: what are these debugs for?
cb_cnt is just a static int..
unless it gets updated in an interrupt handler or something....in which case make it volatile

File: tests/drivers/aio/aio_basic_api/testcase.ini [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [test_aio]
 tags = drivers
 build_only = true
-arch_whitelist = x86
-platform_whitelist = quark_se_c1000_devboard
+arch_whitelist = x86 arc
```
Comment (andrewboie) [inline tests/drivers/aio/aio_basic_api/testcase.ini]: you can probably just delete this line since we are using a platform_whitelist

File: tests/drivers/aio/aio_basic_api/src/test_callback.c [hunk 3]
```diff
@@ -1,54 +1,149 @@
 /*
- * Copyright (c) 2016 Intel Corporation
+ * Copyright (c) 2017 Intel Corporation
```
Comment (andrewboie) [inline tests/drivers/aio/aio_basic_api/src/test_callback.c]: in general we don't need to update the copyright year every time we touch a file

File: tests/drivers/aio/aio_basic_api/src/Makefile [hunk 4]
```diff
@@ -1,2 +1,2 @@
 include $(ZEPHYR_BASE)/tests/Makefile.test
-obj-y = main.o test_callback.o
+obj-y += main.o test_callback.o
```
Comment (andrewboie) [inline tests/drivers/aio/aio_basic_api/src/Makefile]: was this change necessary?

Comment (youvedeep) [inline tests/drivers/aio/aio_basic_api/src/Makefile]: Sure I will remove this change.

File: tests/drivers/aio/aio_basic_api/src/test_callback.c (hunk 1 above)
Comment (youvedeep) [inline tests/drivers/aio/aio_basic_api/src/test_callback.c]: I did not touch cb_cnt it was by default static in this file. I will made it volatile int in my next patch version.

File: tests/drivers/aio/aio_basic_api/testcase.ini (hunk 2 above)
Comment (youvedeep) [inline tests/drivers/aio/aio_basic_api/testcase.ini]: sure will do.

File: tests/drivers/aio/aio_basic_api/src/test_callback.c (hunk 1 above)
Comment (youvedeep) [inline tests/drivers/aio/aio_basic_api/src/test_callback.c]: Debug prints were added to to indicate that cb_cnt is getting cleared, here before rise/fall of GPIO.
If not required, we can remove these.

**Comment by youvedeep** [comment]
for : tests: aio: port AIO cases to run on more platforms 
Qiu, PeiyangX <peiyangx.qiu@intel.com> is original author. I have only contributed in this.

**Review by nashif — APPROVED** [review]
